docs/src/index.md

```diff
@@ -0,0 +1,457 @@
+to this rule. `Symbol`s and strings are allowed to be mixed inside DataFramesMeta macros. 
+Consequently, 
+
+```
+@transform(df, y = :A + cols("B"))
+```
+
+will not error even though 
+
+```
+transform(df, [:A, "B"] => (+) => :y)
+```
+
+will error in DataFrames. 
+
+For consistency, this restriction in the input column types also applies to `@with`
+and `@eachrow`. You cannot mix integer column references with `Symbol` or string column 
+references in `@with` and `@eachrow` in any part of the expression, but you can mix 
+`Symbol`s and strings. The following will fail:
+
+```julia
+df = DataFrame(A = 1:3, B = [2, 1, 2])
+@eachrow df begin 
+    :A = cols(2)
+end
+
+@with df begin 
+    cols(1) + cols("A")
+end
+```
+
+while the following will work without error
+
+```julia
+@eachrow df begin 
+    cols(1) = cols(2)
+end
+
+@with df begin 
+    cols(1) + cols(2)
+end
+```
+
+Note that `cols` is *not* a standard Julia function. It is only used to modify the 
+way that macros in DataFramesMeta.jl escape arguments and has no behavior of its own 
+outside of DataFramesMeta macros.
+
+
+# Working with `Symbol`s without referring to columns
+
+To refer to `Symbol`s without aliasing the column in a data frame, use `^`. 
+
+```
+df = DataFrame(x = [1, 1, 2, 2], y = [1, 2, 101, 102]);
+@select(df, :x2 = :x, :x3 = ^(:x))
+```
+
+This rule applies to all DataFramesMeta macros.
+
+## Comparison with `dplyr` and LINQ
+
+A number of functions for operations on DataFrames have been defined.
+Here is a table of equivalents for Hadley's
+[dplyr](https://github.com/hadley/dplyr) and common
+[LINQ](http://en.wikipedia.org/wiki/Language_Integrated_Query)
+functions.
+
+    Julia             dplyr            LINQ
+    ---------------------------------------------
+    @where            filter           Where
+    @transform        mutate           Select (?)
+    @by                                GroupBy
+    groupby           group_by         GroupBy
+    @combine          summarise/do
+    @orderby          arrange          OrderBy
+    @select           select           Select
+
+
+## `@linq` and other chaining macros
+
+There is also a `@linq` macro that supports chaining and all of the
+functionality defined in other macros. Here is an example of `@linq`:
+
+```julia
+df = DataFrame(a = repeat(1:5, outer = 20),
+               b = repeat(["a", "b", "c", "d"], inner = 25),
+               x = repeat(1:20, inner = 5))
+
+x_thread = @linq df |>
+    transform(y = 10 * :x) |>
+    where(:a .> 2) |>
+    by(:b, meanX = mean(:x), meanY = mean(:y)) |>
+    orderby(:meanX) |>
+    select(:meanX, :meanY, var = :b)
+```
+
+Relative to the use of individual macros, chaining looks cleaner and
+more obvious with less noise from `@` symbols. This approach also
+avoids filling up the limited macro name space. The main downside is
+that more magic happens under the hood.
+
+This method is extensible. Here is a comparison of the macro and
```

Comment:
is this comparison or an example how one could write ones own extension?

Reply:
I don't think I want to expose this API formally yet. Its untested, and I don't know if there is a need to someone to write their own extension. I think we should cut this.

Reply:
OK - then I would cut this.